components/script/dom/bindings/global.rs

```diff
@@ -154,13 +156,22 @@ impl<'a> GlobalRef<'a> {
 
     /// `ScriptChan` used to send messages to the event loop of this global's
     /// thread.
-    pub fn dom_manipulation_task_source(&self) -> Box<ScriptChan + Send> {
+    pub fn script_chan(&self) -> Box<ScriptChan + Send> {
         match *self {
-            GlobalRef::Window(ref window) => window.dom_manipulation_task_source(),
+            GlobalRef::Window(ref window) => window.script_chan(),
             GlobalRef::Worker(ref worker) => worker.script_chan(),
         }
     }
 
+    /// `TaskSource` used to queue DOM manipulation messages to the event loop of this global's
+    /// thread.
+    pub fn dom_manipulation_task_source(&self) -> Box<TaskSource<DOMManipulationTaskMsg> + Send> {
+        match *self {
+            GlobalRef::Window(ref window) => window.dom_manipulation_task_source(),
+            GlobalRef::Worker(_) => unimplemented!(),
```

Comment:
This is unimplemented because I have no idea how the `Worker` handles script channels.